src/descriptor.rs

```diff
@@ -148,10 +152,145 @@ impl Descriptor {
+        let type_tag = DescriptorType::OpReturn;
+        let payload_len = payload.len();
+        if payload_len > MAX_OP_RETURN_LEN {
+            Err(DescriptorError::InvalidPayloadLength(payload_len))
+        } else {
+            Ok(Self {
+                type_tag,
+                payload: payload.to_vec(),
+            })
+        }
+    }
+
+    /// Constructs a new [`Descriptor`] from a P2PKH payload.
+    ///
+    /// The payload is expected to be a valid 20-byte hash.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bitcoin_bosd::{Descriptor, DescriptorType};
+    /// let payload = [0u8; 20]; // all zeros, don't use in production
+    /// let desc = Descriptor::new_p2pkh(&payload);
+    /// # assert_eq!(desc.type_tag(), DescriptorType::P2pkh);
+    /// # assert_eq!(desc.payload(), [0u8; 20]);
+    /// ```
+    pub fn new_p2pkh(payload: &[u8; 20]) -> Self {
+        let type_tag = DescriptorType::P2pkh;
+        Self {
+            type_tag,
+            payload: payload.to_vec(),
+        }
+    }
+
+    /// Constructs a new [`Descriptor`] from a P2SH payload.
+    ///
+    /// The payload is expected to be a valid 20-byte hash.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bitcoin_bosd::{Descriptor, DescriptorType};
+    /// let payload = [0u8; 20]; // all zeros, don't use in production
+    /// let desc = Descriptor::new_p2sh(&payload);
+    /// # assert_eq!(desc.type_tag(), DescriptorType::P2sh);
+    /// # assert_eq!(desc.payload(), [0u8; 20]);
+    /// ```
+    pub fn new_p2sh(payload: &[u8; 20]) -> Self {
+        let type_tag = DescriptorType::P2sh;
+        Self {
+            type_tag,
+            payload: payload.to_vec(),
+        }
+    }
+
+    /// Constructs a new [`Descriptor`] from a P2WPKH payload.
+    ///
+    /// The payload is expected to be a valid 20-byte hash.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bitcoin_bosd::{Descriptor, DescriptorType};
+    /// let payload = [0u8; 20]; // all zeros, don't use in production
+    /// let desc = Descriptor::new_p2wpkh(&payload);
+    /// # assert_eq!(desc.type_tag(), DescriptorType::P2wpkh);
+    /// # assert_eq!(desc.payload(), [0u8; 20]);
+    /// ```
+    pub fn new_p2wpkh(payload: &[u8; 20]) -> Self {
+        let type_tag = DescriptorType::P2wpkh;
+        Self {
+            type_tag,
+            payload: payload.to_vec(),
+        }
+    }
+
+    /// Constructs a new [`Descriptor`] from a P2WSH payload.
+    ///
+    /// The payload is expected to be a valid 32-byte hash.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bitcoin_bosd::{Descriptor, DescriptorType};
+    /// let payload = [0u8; 32]; // all zeros, don't use in production
+    /// let desc = Descriptor::new_p2wsh(&payload);
+    /// # assert_eq!(desc.type_tag(), DescriptorType::P2wsh);
+    /// # assert_eq!(desc.payload(), [0u8; 32]);
+    /// ```
+    pub fn new_p2wsh(payload: &[u8; 32]) -> Self {
```

Comment:
Use the corresponding constant.

Reply:
Done in 99923f646f9b5e6bb68b5f6ad47b2f8c17f3ba5c